test/utilities/screenshot.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { existsSync, mkdirSync, readdirSync } from 'fs';
+import { join } from 'path';
+
+export async function saveFailedTestScreenshot(specTitle: string, testTitle: string, error: Error): Promise<void> {
+  const saveDir = join(__dirname, '..', '..', 'screenshots', specTitle, testTitle);
+  console.log(`Test run failed! Failure: ${error?.message}`);
+  console.log(`Saving a screenshot of the failure here: ${saveDir}`);
+  if (!existsSync(saveDir)) {
+    mkdirSync(saveDir, { recursive: true });
+  }
+  const screenshotPath = getScreenshotTitle(saveDir, error);
+  await browser.saveScreenshot(screenshotPath);
+}
+
+/**
+ * Supports multiple runs and indicates when we're receiving the same error message
+ * repeatedly in the same test.
+ * @param saveDir directory where screenshot will be saved
+ * @param error result from the test run; message is an optional param
+ * @returns 
+ */
+function getScreenshotTitle(saveDir: string, error: Error): string {
+  const errorTitle = error?.message ? error.message : 'Unknown error';
+  let sanitizedTestTitle = errorTitle.replace(/[^a-zA-Z0-9 ]/g, "");
```

Comment:
For the regex, `-` needs to be added, and for good measure, I'd add `'` and `"` too.

Reply:
for the regex, I didn't see any dashes in our `step` names, nor quotations. I can't find the doc link to it now, but the reason this was truncated down so harshly was because github actions puts some limitations on what can be saved as part of the uploadArtifact, and I'd rather us be more conservative to get those titles every time than include quotes. (Previously our use of the colon from several `SFDX: ` commands was causing issues!)